### PR TITLE
Add zip_entry_dwrite function

### LINKED ISSUE #100: File hasn't read access after unpacking
I use `zip_entry_write` to write data to archive. Such unpacked file has no any access. On Linux it's a problem when working in console. So need a way to manually set `zip->entry.external_attr`.

### PR DESCRIPTION
**Pull request by drizt**
This is very quick patch to fix #100. Just to show what is wrong and how it can be fixed.

### CODE REVIEW

File: src/zip.h [hunk 1]
```diff
@@ -192,6 +192,20 @@ extern int zip_entry_write(struct zip_t *zip, const void *buf, size_t bufsize);
...
+  Args:
+    zip: zip archive handler.
+    buf: input buffer.
+    bufsize: input buffer size (in bytes).
+    sd_mode: file open mode. 0664 means user read/write, group read/write, all read.
+
+  Returns:
+    The return code - 0 on success, negative number (< 0) on error.
+*/
+extern int zip_entry_dwrite(struct zip_t *zip, const void *buf, size_t bufsize, int st_mode);
```
Comment (kuba--) [inline src/zip.h]: OK, let's rename the function to sth. like: `zip_entry_writemode` or `zip_entry_writewithmode`.

Comment (kuba--) [reply src/zip.h]: Also take a look how it was done for files:
```c
if ((file_stat.st_mode & 0200) == 0) {
    // MS-DOS read-only attribute
    zip->entry.external_attr |= 0x01;
  }
  zip->entry.external_attr |= (mz_uint32)((file_stat.st_mode & 0xFFFF) << 16);
```

**Comment by kuba--** [comment]
Please add some extra tests for buffers, like this one: https://github.com/kuba--/zip/blob/master/test/test.c#L342

**Comment by drizt** [comment]
See https://github.com/kuba--/zip/pull/103

**Pull request closed without merging**